networking_arista/ml2/arista_sec_gp.py

```diff
@@ -493,61 +583,47 @@ def create_acl(self, sg, security_group_ips=None, existing_acls=None):
             return
 
         security_group_id = sg['id']
+        security_group_rules = list(sg['security_group_rules'])
 
-        # TODO: The rules as text give the port numbers as text, so they won't match our numeric ports
-        known_ingress = set()
-        known_egress = set()
-
-        if existing_acls is not None:
-            for item in six.itervalues(existing_acls):
-                for acl in item.get(self._arista_acl_name(security_group_id, 'ingress'), []):
-                    if 'action' in acl and acl['text'] not in self.aclCreateDict['create']:
-                        known_ingress.add(acl['text'])
-                for acl in item.get(self._arista_acl_name(security_group_id, 'egress'), []):
-                    if 'action' in acl and acl['text'] not in self.aclCreateDict['create']:
-                        known_egress.add(acl['text'])
-
-        in_cmds, out_cmds = self._create_acl_shell(security_group_id)
+        self._sg_enable_dhcp(security_group_rules)
 
-        in_prefix = len(in_cmds)
-        out_prefix = len(out_cmds)
+        cmds = {'ingress': [], 'egress': []}
+        for sgr in security_group_rules:
+            cmds['ingress'], cmds['egress'] = self._create_acl_rule(
+                cmds['ingress'], cmds['egress'], sgr,
+                security_group_ips=security_group_ips
+            )
 
-        security_group_rules = list(sg['security_group_rules'])
-        security_group_rules.append({'protocol': 'dhcp',
-                                     'remote_ip_prefix': None,
-                                     'remote_group_id': None,
-                                     'port_range_min': 'bootps',
-                                     'port_range_max': 'bootpc',
-                                     'direction': 'both'
-                                     })
+        num_rules = {'ingress': len(cmds['ingress']) - 2, 'egress': len(cmds['egress']) - 2}
 
-        for sgr in security_group_rules:
-            in_start = len(in_cmds)
-            out_start = len(out_cmds)
-            in_cmds, out_cmds = self._create_acl_rule(in_cmds, out_cmds, sgr,
-                                                      security_group_ips=security_group_ips)
-            for cmd in in_cmds[in_start:]:
-                known_ingress.discard(cmd)
-            for cmd in out_cmds[out_start:]:
-                known_egress.discard(cmd)
-
-        num_rules = { 'ingress': len(in_cmds) - in_prefix, 'egress': len(out_cmds) - out_prefix }
-        in_cmds = in_cmds[:in_prefix] + [str('no ' + rule) for rule in known_ingress] + in_cmds[in_prefix:]
-        out_cmds = out_cmds[:out_prefix] + [str('no ' + rule) for rule in known_egress] + out_cmds[out_prefix:]
-
-        in_cmds.append('exit')
-        out_cmds.append('exit')
+        # Try consolidation
+        if 0 < self.max_rules < num_rules['ingress'] + num_rules['egress']:
```

Comment:
As I read it, the consolidation is "lossless", so the resulting rules are effectively equivalent, just more compact. This should be done all the time.

But we also need a "lossy" compaction, so that we never get over the limit.